keps/sig-node/3542-cri-image-pulling-with-progress-notification/README.md

```diff
@@ -0,0 +1,876 @@
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Rollout should not fail copmletely, it should fallback to existing stable PullImage call in case
+the runtime does not support PullImageWithProgress call.
+
+Rollback is disabling feature in kubelet config, no changes aside from kubelet config should be
+needed.
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+###### What specific metrics should inform a rollback?
+
+Runtime reporting callback is not implemented shold signify need for fallback to regular PullImage.
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+This is not workload-related feature.
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+###### How can someone using this feature know that it is working for their instance?
+
+`kubectl describe pods/pod` should in events section show that:
+
+- [required] Image pulling has started, with progress reporting
+- [optional] Image pulling is at particular point
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+In Alpha:
+
+- image pulling operation should fail faster.
```

Comment:
large image pulls don't timeout on CRI timeout and are completed without being restarted/cancelled

Reply:
I don't think this was about image size. Wasn't the point of no-progress timeout to fail faster when there is no progress and the image is impossible to download? I'll make it clear in phrasing here.